components/family-studies/migrations/src/main/java/org/phenotips/studies/family/migrations/R71498PhenoTips2155DataMigration.java

```diff
@@ -132,264 +202,638 @@ public Object doInHibernate(Session hSession) throws HibernateException, XWikiEx
             Query q =
                 this.session.createQuery("select distinct o.name from BaseObject o, BaseObject ro where"
                     + " o.className = :patclass and o.name <> 'PhenoTips.PatientTemplate'"
-                    + " and ro.name = o.name and ro.className = :relclass");
```

Comment:
Then remove `ro` completely from the query, it doesn't do anything except slow things down.